File: futter/front/heyiisched/lib/models/emploi.dart


class Emploi {
  final String idEmploi;
  final String typeEmploi;
  final String jour;
  final String heureDebut;
  final String heureFin;
  final List<dynamic> enseignants;
  final List<dynamic> salles;
  final List<dynamic> grpClasses;
  final dynamic admin;
  final dynamic cours;

  Emploi({
    required this.idEmploi,
    required this.typeEmploi,
    required this.jour,
    required this.heureDebut,
    required this.heureFin,
    required this.enseignants,
    required this.salles,
    required this.grpClasses,
    this.admin,
    this.cours,
  });

  factory Emploi.fromJson(Map<String, dynamic> json) {
    return Emploi(
      idEmploi: json['idEmploi'] ?? '',
      typeEmploi: json['typeEmploi'] ?? '',
      jour: json['jour'] ?? '',
      heureDebut: json['heureDebut'] ?? '',
      heureFin: json['heureFin'] ?? '',
      enseignants: json['enseignants'] ?? [],
      salles: json['salles'] ?? [],
      grpClasses: json['grpClasses'] ?? [],
      admin: json['admin'] ?? '',
      cours: json['cours'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'idEmploi': idEmploi,
      'typeEmploi': typeEmploi,
      'jour': jour,
      'heureDebut': heureDebut,
      'heureFin': heureFin,
      'enseignants': enseignants,
      'salles': salles,
      'grpClasses': grpClasses,
      'admin': admin,
      'cours': cours,
    };
  }
}

File: futter/back/heyii-Vhouyem/src/main/java/com/example/heyii/Entity/Emploi.java
package com.example.heyii.Entity;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.io.Serializable;
import java.time.LocalTime;
import java.util.List;

@Getter
@Setter
@ToString
@Document(collection = "emploi")
public class Emploi implements Serializable {

    @Id
    private String idEmploi;

    private String typeEmploi; // enseignant, étudiant, salle
    private String jour;
    private LocalTime heureDebut;
    private LocalTime heureFin;

    @DBRef  // Référence MongoDB pour la relation
    private List<Enseignant> enseignants;

    @DBRef  // Référence MongoDB pour la relation
    private List<Salle> salles;

    @DBRef  // Référence MongoDB pour la relation
    private List<GrpClass> grpClasses;

    @DBRef  // Référence MongoDB pour la relation
    private Admin admin;

    @DBRef  // Référence MongoDB pour la relation
    private List<Cours> cours;

    // Constructeur par défaut
    public Emploi() {}

    // Constructeur avec paramètres
    public Emploi(String idEmploi) {
        this.idEmploi = idEmploi;
    }

    // Constructeur avec paramètres
    public Emploi(String typeEmploi, String jour, LocalTime heureDebut, LocalTime heureFin,
                  List<Enseignant> enseignants, List<GrpClass> grpClasses, List<Salle> salles) {
        this.typeEmploi = typeEmploi;
        this.jour = jour;
        this.heureDebut = heureDebut;
        this.heureFin = heureFin;
        this.enseignants = enseignants;
        this.grpClasses = grpClasses;
        this.salles = salles;
    }
}
